common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -356,6 +375,23 @@ public boolean close() {
         }
 
         @Override
+        public boolean close(T trackedObject) {
+            assert referentHash == System.identityHashCode(trackedObject);
+
+            // We need to actually do the null check of the trackedObject after we call leak.close() because otherwise
+            // we may get false-positives reported by the ResourceLeakDetector. This can happen as the JIT / GC may
+            // be able to figure out that we do not need the trackedObject anymore and so already enqueue it for
+            // collection before we actually get a chance to close the enclosing ResourceLeak.
+            //
+            // A more proper fix in the future is to change the ResourceLeak.close() method to take the tracked object
+            // as and so ensure it will not be collected before we had a chance to close the ResourceLeak.
```

Comment:
@Scottmitch I need to adjust the comment here if you agree how I fixed it now makes sense in terms of API changes.